website/docs/r/cdn_frontdoor_custom_domain.html.markdown

```diff
@@ -0,0 +1,135 @@
+  resource_group_name = azurerm_resource_group.example.name
+  ttl                 = 3600
+
+  record {
+    value = azurerm_cdn_frontdoor_custom_domain.example.validation_token
+  }
+}
+```
+
+## Example CNAME Record Usage
+
+!>**IMPORTANT:** You **must** include the `depends_on` meta-argument which references both the `azurerm_cdn_frontdoor_route` and the `azurerm_cdn_frontdoor_security_policy` that are associated with your Custom Domain.
+
+```hcl
+resource "azurerm_dns_cname_record" "example" {
+  depends_on = [azurerm_cdn_frontdoor_route.example, azurerm_cdn_frontdoor_security_policy.example]
+
+  name                = "contoso"
+  zone_name           = azurerm_dns_zone.example.name
+  resource_group_name = azurerm_resource_group.example.name
+  ttl                 = 3600
+  record              = azurerm_cdn_frontdoor_endpoint.example.host_name
+}
+```
+
+## Arguments Reference
+
+* `name` - (Required) The name which should be used for this Frontdoor Custom Domain. Possible values must be between 2 and 260 characters in length, must begin with a letter or number, end with a letter or number and contain only letters, numbers and hyphens. Changing this forces a new Frontdoor Custom Domain to be created.
+
+* `cdn_frontdoor_profile_id` - (Required) The ID of the Frontdoor Profile. Changing this forces a new Frontdoor Profile to be created.
+
+* `host_name` - (Required) The host name of the domain. Changing this forces a new Frontdoor Custom Domain to be created.
+
+* `dns_zone_id` - (Optional) The Resource ID of the DNS Zone that is to be used for the Frontdoor Custom Domain.
+
+* `pre_validated_cdn_frontdoor_custom_domain_id` - (Optional) The resource ID of the pre-validated Frontdoor Custom Domain. This domain type is used when you wish to onboard a validated Azure service domain, and then configure the Azure service behind an Azure Front Door.
+
+->**NOTE:** Currently `pre_validated_cdn_frontdoor_custom_domain_id` only supports domains validated by Static Web App.
+
+* `tls` - (Required) A `tls` block as defined below.
+
+---
+
+A `tls` block supports the following:
+
+* `certificate_type` - (Optional) Defines the source of the SSL certificate. Possible values include `CustomerCertificate` and `ManagedCertificate`. Defaults to `ManagedCertificate`.
+
+->**NOTE:** It may take upto 15 minutes for the Frontdoor Service to validate the state and Domain ownership of the Custom Domain.
+
+* `minimum_tls_version` - (Optional) TLS protocol version that will be used for Https. Possible values include `TLS10` and `TLS12`. Defaults to `TLS12`.
+
+* `cdn_frontdoor_secret_id` - (Optional) Resource ID of the Frontdoor Secrect.
+
+---
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported:
+
+* `id` - The ID of the Frontdoor Custom Domain.
+
+* `domain_validation_state` - Provisioning substate shows the progress of custom HTTPS enabling/disabling process.
```

Comment:
(as above) since this isn't something that's really usable, can we remove this?

```suggestion
```

Reply:
Since the field has been removed from the resource I have also removed it from the documentation. Fixed.